airflow/hooks/hive_hooks.py

```diff
@@ -878,6 +878,7 @@ def to_csv(
         header = next(results_iter)
         message = None
 
+        i = 0
```

Comment:
if log error, shouldn't we change L904 instead? If we set i = 0, then it will get shadowed in L892.

Reply:
We could not change L904 due to we don't know if `i` exists or not. set `i = 0` just init `i`.
If hql statement return no reocrd, L892 will not run and `i = 0`
else hql statement have record and `i` is reocrd number.

Reply:
@feng-tao Of course we can write some code like
```python
self.log.info("Done. Loaded a total of %s rows.", i if 'i' in locals() else 0)
```
but I think is a little strange, so I put `i = 0` in L881

Reply:
Yeah doming something like that is a bit odd.

One question though: does this change work on Python3? I have half-memories of the scoping rules for `for` being different in py2 and py3, but I can't remember how.

Reply:
@ashb I run some code below in Python2.7.15 and Python3.6.7, both of them ware asserted
```python
i = 0
lst = list(range(10))
for i, v in enumerate(lst, 1):
    pass
assert i == len(lst), "counter {} not equel to len {}".format(i, len(lst))
```
Which mean that work on py2 and py3.

But I found some thing wrong when I write test code in 
https://github.com/apache/airflow/blob/c8891454fd95ca21d5131ff26e597f475e4f54f6/airflow/hooks/hive_hooks.py#L891-L894
We should use `enumerate(results_iter, 1)` instead of `enumerate(results_iter, 1)` to count the correct row number. Am I right?

So I should fix it in this PR or submit new PR to do that?

Reply:
@ashb @feng-tao PTAL, thx